oidc_test.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func TestGetToken(t *testing.T) {
+
+	var response string
+	code := 200
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(code)
+		fmt.Fprintln(w, response)
+	}))
+	defer ts.Close()
+	p := oidcProvider{
+		client:       &http.Client{},
+		tokenUrl:     ts.URL,
+		clientId:     "testid",
+		clientSecret: "testsecret",
+	}
+
+	ctx := context.Background()
+
+	tcs := []struct {
+		input string
+		code  int
+		fail  bool
+	}{
+		{
+			input: `{"access_token": "aaa"}`,
+			code:  200,
+			fail:  false,
+		},
+		{
```

Comment:
May I suggest an alternative structuring? It makes easier to understand which test failed.
```suggestion
	tcs := map[string]struct {
		input string
		code  int
		fail  bool
	}{
		"GivenValidAccessToken_ThenExpectStatus200": {
			input: `{"access_token": "aaa"}`,
			code:  200,
			fail:  false,
		},
		{
```